### PR TITLE
Background auth validation fix

### PR DESCRIPTION
**Pull request by gapra-msft**
… twice

### CODE REVIEW

File: cmd/mount.go [hunk 1]
```diff
@@ -293,13 +293,17 @@ var mountCmd = &cobra.Command{
...
-			log.Err("Mount: error initializing new pipeline [%v]", err)
-			fmt.Println("failed to mount :", err)
-			Destroy(1)
+		var pipeline *internal.Pipeline
+
+		if !daemon.WasReborn() {
+			log.Crit("Starting Blobfuse2 Mount : %s on (%s)", common.Blobfuse2Version, common.GetCurrentDistro())
+			log.Crit("Logging level set to : %s", logLevel.String())
+			pipeline, err = internal.NewPipeline(options.Components)
+			if err != nil {
```
Comment (vibhansa-msft) [inline cmd/mount.go:302]: newPipeline we need to do in background as well. Only the auth part we wanted to shut down. How does this work ?

Comment (gapra-msft) [reply cmd/mount.go:302]: Why? Doesnt the old pipeline variable get inherited?

Comment (gapra-msft) [reply cmd/mount.go:302]: nvm youre right. Let me look into this again

File: internal/base_component.go [hunk 2]
```diff
@@ -62,7 +62,7 @@ func (base *BaseComponent) SetName(name string) {
 	base.compName = name
 }
 
-func (base *BaseComponent) Configure() error {
+func (base *BaseComponent) Configure(validateCredentials bool) error {
```
Comment (vibhansa-msft) [inline internal/base_component.go:65]: validateCredential does not make sense for all other components, so better to call this something which says "its running in background/daemon mode"

File: internal/pipeline.go [hunk 3]
```diff
@@ -52,7 +52,7 @@ type NewComponent func() Component
 var registeredComponents map[string]NewComponent
 
 // NewPipeline : Using a list of strings holding name of components, create and configure the component objects
-func NewPipeline(components []string) (*Pipeline, error) {
+func NewPipeline(components []string, validateCredentials bool) (*Pipeline, error) {
```
Comment (vibhansa-msft) [inline internal/pipeline.go:55]: same here, no validate-creds just isDaemon or something like that.

File: component/libfuse/libfuse_handler_test_wrapper.go [hunk 4]
```diff
@@ -77,7 +77,7 @@ func newTestLibfuse(next internal.Component, configuration string) *Libfuse {
 	config.ReadConfigFromReader(strings.NewReader(configuration))
 	libfuse := NewLibfuseComponent()
 	libfuse.SetNextComponent(next)
-	libfuse.Configure()
+	libfuse.Configure(true)
```
Comment (vibhansa-msft) [inline component/libfuse/libfuse_handler_test_wrapper.go:80]: Similar changes are needed for fuse2 as well, it's not building right now.

Comment (gapra-msft) [reply component/libfuse/libfuse_handler_test_wrapper.go:80]: good catch

File: internal/pipeline.go (hunk 3 above)
Comment (gapra-msft) [reply internal/pipeline.go]: renamed to isParent

File: internal/base_component.go (hunk 2 above)
Comment (gapra-msft) [reply internal/base_component.go]: done

**Review by souravgupta-msft — APPROVED** [review]

File: internal/pipeline.go [hunk 5]
```diff
@@ -63,7 +63,7 @@ func NewPipeline(components []string) (*Pipeline, error) {
 			comp := compInit()
 
 			// request component to parse and validate config of its interest
-			err := comp.Configure()
+			err := comp.Configure(isParent)
```
Comment (vibhansa-msft) [inline internal/pipeline.go:66]: Does isParent makes more sense in terms of name of isDaemon makes it more obvious.

**Review by vibhansa-msft — APPROVED** [review]

File: internal/pipeline.go (hunk 5 above)
Comment (gapra-msft) [reply internal/pipeline.go:66]: that would make it a negative bool cause we only want to run it in the Parent code not the Daemon code

Comment (gapra-msft) [reply internal/pipeline.go:66]: Hopefully adding this comment in azstorage where we consume isParent bool helps
// The daemon runs all pipeline Configure code twice. isParent allows us to only validate credentials in parent mode, preventing a second unnecessary REST call.

**Pull request merged**